tests/tests/test-tx-index-bound-check.ts

```diff
@@ -0,0 +1,23 @@
+import { expect } from "chai";
+import { createTransfer } from "../util/transactions";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+
+describeDevMoonbeam("Transaction Index", (context) => {
+  it("should get transaction by index", async function () {
+    const block = 1;
+    const index = 0;
+    const testAccount = "0x1111111111111111111111111111111111111111";
+    await context.createBlock({
+      transactions: [await createTransfer(context, testAccount, 0)],
+    });
+    let result = await context.web3.eth.getTransactionFromBlock(block, index);
+    expect(result.transactionIndex).to.equal(index);
+  });
+  it("should return out of bounds message", async function () {
+    const block = 0;
+    const index = 0;
+    await context.web3.eth
+      .getTransactionFromBlock(block, index)
+      .catch((err) => expect(err.message).to.equal(`Returned error: ${index} is out of bounds`));
```

Comment:
If this didn't throw an error for some reason, I think this test would pass. I think there's a way to expect `getTransactionFromBlock()` to error, or perhaps you could add another case that fails if the call succeeds...